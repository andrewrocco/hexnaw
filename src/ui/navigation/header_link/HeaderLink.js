import React from 'react';
import PropTypes from 'prop-types';
import Link from 'next/link';

import { StyledLink } from './HeaderLink.styles';

export const HeaderLink = ({ children, href, prefetch }) => (
  <Link href={href} prefetch={prefetch} passHref>
    <StyledLink>{children}</StyledLink>
  </Link>
);

HeaderLink.propTypes = {
  children: PropTypes.node,
  href: PropTypes.string,
  prefetch: PropTypes.bool,
};
