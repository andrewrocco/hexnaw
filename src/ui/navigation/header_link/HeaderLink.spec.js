import React from 'react';

import { HeaderLink } from './HeaderLink';

describe('HeaderLink Component', () => {
  context('when nothing specified', () => {
    it('should render correctly', () => {
      const subject = shallow(<HeaderLink>Click Me</HeaderLink>);
      expect(subject).toMatchSnapshot();
    });
  });
});
