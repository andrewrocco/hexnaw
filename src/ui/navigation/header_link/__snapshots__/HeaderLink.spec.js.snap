// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderLink Component when nothing specified should render correctly 1`] = `
<Link
  passHref={true}
>
  <styled.a>
    Click Me
  </styled.a>
</Link>
`;
